### ISSUE TITLE
In the logits design phase, does the bias get applied to both terms in Eq (1)?

### ISSUE THREAD

**Issue by tsjain** [manifoldbio/mber-open#12]
I had a question on the application of the bias term from ESM to the design logits. From equation 1 in the Supplementary Info, the goal is to apply the PLM logits to the design vector being modified within ColabDesign.

However, looking at the soft_seq function in ColabDesign, it seems like the bias is only added to the second term (under the softmax operation), while the first term still uses the backpropagated values scaled by opt["alpha"].

Sorry, if I've misunderstood the code.

**Comment by eswan01** [comment]
Thanks for raising this issue. Taking a look at the [ColabDesign code](https://github.com/sokrypton/ColabDesign/blob/e31a56fe1d9b4de25c8697f3a28b75892941cc72/colabdesign/shared/model.py#L212-L223), you are right and the bias is only added to the softmaxed term, while `seq["input"]` is used in place of `seq["logits"]` for the linearly scaled logits:

```
  # straight-through/reparameterization
  seq["logits"] = seq["input"] * opt["alpha"]
  if bias is not None: seq["logits"] = seq["logits"] + bias
  seq["pssm"] = jax.nn.softmax(seq["logits"])
  seq["soft"] = jax.nn.softmax(seq["logits"] / opt["temp"])
  seq["hard"] = jax.nn.one_hot(seq["soft"].argmax(-1), seq["soft"].shape[-1])
  seq["hard"] = jax.lax.stop_gradient(seq["hard"] - seq["soft"]) + seq["soft"]


  # create pseudo sequence
  seq["pseudo"] = opt["soft"] * seq["soft"] + (1-opt["soft"]) * seq["input"]
  seq["pseudo"] = opt["hard"] * seq["hard"] + (1-opt["hard"]) * seq["pseudo"]
  return seq
```

In mBER, we do still apply the bias initially, but as a normalized probability vector, rather than the raw bias logits ([code](https://github.com/manifoldbio/mber-open/blob/fb3e6ae00b6d1e71a326ad4651dbc5f635dbf5f0/src/mber/core/modules/trajectory.py#L302-L310)).

We'll update the supplement in a future version of the preprint to more accurately reflect the adding of the bias term.

**Comment by tsjain** [comment]
Thanks for your quick response and clarification.
